03_ocp_repo_sync.sh

```diff
@@ -49,3 +49,21 @@ pushd "${GOPATH}/src/github.com/metalkube/facet"
 yarn install
 ./build.sh
 popd
+
+# Install Go dependency management tool
+# Using pre-compiled binaries instead of installing from source
+curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+
+
+# Install operator-sdk for use by the baremetal-operator
+sync_go_repo_and_patch github.com/operator-framework/operator-sdk https://github.com/operator-framework/operator-sdk.git
+
+# Build operator-sdk
+pushd "${GOPATH}/src/github.com/operator-framework/operator-sdk"
+git checkout master
+make dep
```

Comment:
This currently doesn't work because we're not installing dep:

```
+ pushd /home/shardy/go/src/github.com/operator-framework/operator-sdk
~/go/src/github.com/operator-framework/operator-sdk ~/metalkube/dev-scripts
+ git checkout master
Switched to branch 'master'
+ make dep
make[1]: Entering directory `/home/shardy/go/src/github.com/operator-framework/operator-sdk'
make[1]: dep: Command not found
make[1]: *** [dep] Error 127
make[1]: Leaving directory `/home/shardy/go/src/github.com/operator-framework/operator-sdk'
make: *** [repo_sync] Error 2
```

@imain pushed https://github.com/metalkube/dev-scripts/pull/103 to fix that, but I made some comments re how it handles GOPATH.

So I guess we either merge that then fix the path handling, or you could squash that commit into this PR and address my comment

Reply:
Actually even with #103 this still fails because the operator-sdk makefile expects dep to be on the path, I had to do:

```
[shardy@tripleodev2 operator-sdk]$ eval $(go env)
[shardy@tripleodev2 operator-sdk]$ echo $GOPATH 
/home/shardy/go
[shardy@tripleodev2 operator-sdk]$ export PATH=${GOPATH}/bin:$PATH
[shardy@tripleodev2 operator-sdk]$ which dep
~/go/bin/dep
```

So we need to do similar in the script I think.

Reply:
Script has been updated.

Reply:
Made the change to fix the build error.